src/theory/logic_info.cpp

```diff
@@ -275,14 +284,26 @@ std::string LogicInfo::getLogicString() const {
     LogicInfo qf_all_supported;
     qf_all_supported.disableQuantifiers();
     qf_all_supported.lock();
-    if(hasEverything()) {
+    if (hasEverything())
+    {
       d_logicString = "ALL";
-    } else if(*this == qf_all_supported) {
+    }
+    else if (hasEverything(true))
```

Comment:
I don't think we need this distinction.  Just print "HO_" if higher order, then "ALL" if hasEverything.

Reply:
Then I have to change the control flow of the whole method. The way it is seems harmless to me.

Reply:
The control flow is getting much more confusing in this PR. Let's keep it simple.

Reply:
Alright, I've changed it so that it's done like you suggested.